File: spring-webflux/src/main/java/victor/training/reactive/usecase/migrating/MigrationApp.java
package victor.training.reactive.usecase.migrating;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.core.publisher.Mono;

@RestController
@SpringBootApplication
public class MigrationApp {
   public static void main(String[] args) {
//      Utils.installBlockHound();
      SpringApplication.run(MigrationApp.class, args);
   }

   @GetMapping
   public Mono<String> reactiveEndpoint(@RequestParam(defaultValue = "1") String id) {
      return WebClient.create().get().uri("localhost:8080/call").retrieve()
          .bodyToMono(String.class) // on netty scheduler
          // .delayElement(Duration.ofMillis(1)) // moves on parallel
          .doOnNext(response -> SomeLibrary.validate(response))
          .map(String::toUpperCase);
   }

   @GetMapping
   public Mono<String> call() {
      return Mono.just("data");
   }
}

File: spring-webflux/src/main/java/victor/training/reactive/usecase/migrating/MigrationApp.kt
package victor.training.reactive.usecase.migrating

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.reactive.function.client.WebClient
import reactor.core.publisher.Mono
import java.util.*

@RestController
@SpringBootApplication
class MigrationApp {
    @GetMapping
    fun reactiveEndpoint(@RequestParam(defaultValue = "1") id: String?): Mono<String> {
        return WebClient.create().get().uri("localhost:8080/call").retrieve()
            .bodyToMono(String::class.java) // on netty scheduler
            // .delayElement(Duration.ofMillis(1)) // moves on parallel
            .doOnNext { response: String? -> SomeLibrary.validate(response) }
            .map { obj: String -> obj.uppercase(Locale.getDefault()) }
    }

    @GetMapping
    fun call(): Mono<String> {
        return Mono.just("data")
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
//      Utils.installBlockHound();
            SpringApplication.run(MigrationApp::class.java, *args)
        }
    }
}